src/models/Event.js

```diff
@@ -0,0 +1,53 @@
+const Group = require("./Group");
+const mongoose = require("mongoose");
+
+const eventSchema = new mongoose.Schema(
+  {
+    groupID: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Group",
+    },
+    name: {
+      type: String,
+      required: [true, "please provide an activity name!"],
+    },
+    description: {
+      type: String,
+      default: "a default event description",
+    },
+    eventDateTime: {
+      type: Date,
+    },
+    host: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User",
+    },
+    participants: [
+      {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+      },
+    ],
+    activities: [
+      { type: mongoose.Schema.Types.ObjectId, ref: "EventActivity" },
+    ],
+    chosenActivity: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "EventActivity",
+    },
+  },
+  { timestamps: true }
+);
+
+// Middleware to add the newly created event's _id into groupEvents in the Group schema
+eventSchema.pre("save", async function (next) {
+  const foundGroup = await Group.findOne({ _id: this.groupID });
+
+  foundGroup.groupEvents.push(this._id);
```

Comment:
When I create a new group, it doesn't have a `groupEvents` property, so I get a null reference error here. I'm wondering, actually, if we need the association data between groups and events in both places. (If we were using a relational DB like Postgres or MySQL, we generally would not do this.) See my [comment here ](https://github.com/Code-the-Dream-School/ee-prac-team2-back/pull/55/files#r1424394368) on Marice's PR about this too. Maybe we can discuss in slack.

Reply:
Yes this is because of this [bug](https://github.com/Code-the-Dream-School/ee-prac-team2-back/pull/55#discussion_r1422858986) from the group data model PR. I have this bug fixed, and have tested this out locally without any errors. In a mongoose schema, if a referenced field is declared as an array, on document creation an empty field is automatically added for that referenced array field. So once @marice-romero fixes that bug, there shouldn't be any errors for this.

And as for the reason of having association of events in groups, this kind of returns to my thinking answered in [here](https://github.com/Code-the-Dream-School/ee-prac-team2-back/pull/55#discussion_r1424498197)